persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/configuration/JdbcStringBasedStoreConfigurationBuilder.java

```diff
@@ -99,6 +101,23 @@ public StringTableManipulationConfigurationBuilder table() {
       return table;
    }
 
+   @Override
+   public void validate() {
+      if (connectionFactory != null) {
+         connectionFactory.validate();
+      }
+      // TODO: I don't think this is quite correct...
+      if (attributes.attribute(MANAGE_CONNECTION_FACTORY).get() && (connectionFactory == null || connectionFactory.needsManaging())) {
```

Comment:
Can't we do this without the need for the `needsManaging` interface. The following should be sufficient:

```java
if (!attributes.attribute(MANAGE_CONNECTION_FACTORY).get())
   table.validate()
```

If we add `builder.manageConnectionFactory(false);` to the `ManagedConnectionFactoryConfigurationBuilder` constructor.